tests/integration/operatorhub/cmd_service_test.go

```diff
@@ -29,521 +29,514 @@ var _ = Describe("odo service command tests for OperatorHub", func() {
+
+							When("odo push is executed", func() {
+								JustBeforeEach(func() {
+									helper.CmdShouldPass("odo", "push")
+								})
+
+								JustAfterEach(func() {
+								})
+
+								It("should create pods in running state", func() {
+									// now verify if the pods for the operator have started
+									pods := oc.GetAllPodsInNs(commonVar.Project)
+									// Look for pod with example name because that's the name etcd will give to the pods.
+									etcdPod := regexp.MustCompile(`example-.[a-z0-9]*`).FindString(pods)
+
+									ocArgs := []string{"get", "pods", etcdPod, "-o", "template=\"{{.status.phase}}\"", "-n", commonVar.Project}
+									helper.WaitForCmdOut("oc", ocArgs, 1, true, func(output string) bool {
+										return strings.Contains(output, "Running")
+									})
+
+								})
+							})
+						})
+
+						When("a service is created from the output of the dryRun command with a specific name", func() {
+
+							var name string
+							var svcFullName string
+							JustBeforeEach(func() {
+								name = helper.RandString(6)
+								svcFullName = strings.Join([]string{"EtcdCluster", name}, "/")
+								helper.CmdShouldPass("odo", "service", "create", "--from-file", fileName, name, "--project", commonVar.Project)
+							})
+
+							JustAfterEach(func() {
+								helper.CmdShouldPass("odo", "service", "delete", svcFullName, "-f")
+							})
+
+							When("odo push is executed", func() {
+
+								JustBeforeEach(func() {
+									helper.CmdShouldPass("odo", "push")
+								})
+
+								JustAfterEach(func() {
+								})
+
+								It("should fail to create a service again with the same name", func() {
+									stdOut = helper.CmdShouldFail("odo", "service", "create", "--from-file", fileName, name, "--project", commonVar.Project)
+									Expect(stdOut).To(ContainSubstring("please provide a different name or delete the existing service first"))
+								})
+
+								It("should create pods in running state", func() {
+									// now verify if the pods for the operator have started
+									pods := oc.GetAllPodsInNs(commonVar.Project)
+									// Look for pod with example name because that's the name etcd will give to the pods.
+									etcdPod := regexp.MustCompile(name + `-.[a-z0-9]*`).FindString(pods)
+
+									ocArgs := []string{"get", "pods", etcdPod, "-o", "template=\"{{.status.phase}}\"", "-n", commonVar.Project}
+									helper.WaitForCmdOut("oc", ocArgs, 1, true, func(output string) bool {
+										return strings.Contains(output, "Running")
+									})
+
+								})
+							})
+						})
+					})
+				})
+
+				When("an EtcdCluster instance is created with no name", func() {
+					var stdOut string
+					JustBeforeEach(func() {
+						stdOut = helper.CmdShouldPass("odo", "service", "create", fmt.Sprintf("%s/EtcdCluster", etcdOperator), "--project", commonVar.Project)
+					})
+
+					JustAfterEach(func() {
+					})
+
+					It("should create the EtcdCluster instance", func() {
```

Comment:
This isn't really creating the instance of the Operator backed service. It just adds the configuration for that into devfile.yaml; which is something we're testing next.

Would it make sense to combine the two specs here? 🤔

Reply:
To be consistent with the other test, yes. good catch